src/components/AuthForm.tsx

```diff
@@ -0,0 +1,269 @@
+      }
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className={classes.form}>
+      {!isSignIn && (
+        <div>
+          <input
+            type="text"
+            id="username"
+            name="username"
+            placeholder="Username"
+            value={authData.username}
+            onChange={handleInputChange}
+            onBlur={handleBlur}
+            className={classes["form-input"]}
+          />
+          {errors.map(
+            (error) =>
+              error.fieldName === "username" && (
+                <div
+                  key={error.fieldName}
+                  className={classes["form-input__error"]}
+                >
+                  {error.message}
+                </div>
+              )
+          )}
+        </div>
+      )}
+      <div>
+        <input
+          type="email"
+          id="email"
+          name="email"
+          placeholder="Email"
+          value={authData.email}
+          onChange={handleInputChange}
+          onBlur={handleBlur}
+          className={classes["form-input"]}
+        />
+        {errors.map(
+          (error) =>
+            error.fieldName === "email" && (
+              <div
+                key={error.fieldName}
+                className={classes["form-input__error"]}
+              >
+                {error.message}
+              </div>
+            )
+        )}
+      </div>
+      <div>
+        <input
+          type="password"
+          id="password"
+          name="password"
+          placeholder="Password"
+          value={authData.password}
+          onChange={handleInputChange}
+          onBlur={handleBlur}
+          className={classes["form-input"]}
+        />
+        {errors.map(
+          (error) =>
+            error.fieldName === "password" && (
+              <div
+                key={error.fieldName}
+                className={classes["form-input__error"]}
+              >
+                {error.message}
+              </div>
+            )
+        )}
+      </div>
+      <div>
+        {!isSignIn && (
+          <div>
+            <input
+              type="password"
+              id="confirmPassword"
+              name="confirmPassword"
+              placeholder="Confirm Password"
+              value={authData.confirmPassword}
+              onChange={handleInputChange}
+              onBlur={handleBlur}
+              className={classes["form-input"]}
+            />
+            {errors.map(
+              (error) =>
+                error.fieldName === "confirmPassword" && (
+                  <div
+                    key={error.fieldName}
+                    className={classes["form-input__error"]}
+                  >
+                    {error.message}
+                  </div>
+                )
+            )}
+          </div>
+        )}
+      </div>
+      <button type="submit" className={classes["form-btn__submit"]}>
+        {isSignIn ? "Sign In" : "Sign Up"}
```

Comment:
one side note, not that this is the case, but when on the topic of clean code, never nest ternary expressions.